django/thunderstore/repository/api/experimental/tests/test_async_submit.py

```diff
@@ -0,0 +1,277 @@
+from django.urls import reverse
+from rest_framework.test import APIClient
+
+from thunderstore.community.factories import CommunityFactory, PackageCategoryFactory
+from thunderstore.community.models import Community, PackageCategory, PackageListing
+from thunderstore.core.factories import UserFactory
+from thunderstore.core.types import UserType
+from thunderstore.frontend.api.experimental.serializers.views import (
+    CommunitySerializer,
+    PackageCategorySerializer,
+)
+from thunderstore.repository.models import Team, TeamMember, TeamMemberRole
+from thunderstore.repository.package_reference import PackageReference
+from thunderstore.usermedia.models import UserMedia
+
+
+def _handle_submit(api_client: APIClient, data: str):
+    submission = api_client.post(
+        reverse("api:experimental:submission.submit-async"),
+        data,
+        content_type="application/json",
+    )
+    if not submission.status_code == 200:
+        return submission
+    return api_client.get(
+        reverse(
+            "api:experimental:submission.poll-async",
+            kwargs={
+                "submission_id": submission.json()["id"],
+            },
+        ),
+        content_type="application/json",
+    )
+
+
+@pytest.mark.django_db(transaction=True)
+def test_api_experimental_submit_package_async_success(
+    api_client: APIClient,
+    user: UserType,
+    manifest_v1_data: Dict[str, Any],
+    package_category: PackageCategory,
+    team: Team,
+    community: Community,
+    manifest_v1_package_upload_id: str,
+):
+    com2 = CommunityFactory()
+    com3 = CommunityFactory()
+    com2_cat1 = PackageCategoryFactory(community=com2)
+    com3_cat1 = PackageCategoryFactory(community=com3)
+    com3_cat2 = PackageCategoryFactory(community=com3)
+
+    TeamMember.objects.create(
+        user=user,
+        team=team,
+        role=TeamMemberRole.owner,
+    )
+
+    api_client.force_authenticate(user=user)
+    response = _handle_submit(
+        api_client,
+        json.dumps(
+            {
+                "upload_uuid": manifest_v1_package_upload_id,
+                "author_name": team.name,
+                "categories": [package_category.slug],
+                "community_categories": {
+                    com2.identifier: [com2_cat1.slug],
+                    com3.identifier: [com3_cat1.slug, com3_cat2.slug],
+                },
+                "communities": [community.identifier, com2.identifier, com3.identifier],
+                "has_nsfw_content": True,
+            }
+        ),
+    )
+    print(response.content)
+    assert response.status_code == 200
+    response_data = response.json()["result"]
+    version_data = response_data["package_version"]
+    assert version_data["namespace"] == team.name
+    assert version_data["name"] == manifest_v1_data["name"]
+    assert version_data["version_number"] == manifest_v1_data["version_number"]
+    assert PackageReference(team.name, "name", "1.0.0").exists is True
```

Comment:
Should the name and version number be read from the manifest to check that the created package actually has the expected values, and not e.g. hardcoded "name"?

Reply:
Would be better yeah. The reason it's like this is because it's copypasta from the non-async versions of the package submission tests, as I was aiming to validate it passes the same tests as that one (I initially attempted to modify the pre-existing tests to run the same test code for both sync & async flow, but that turned out to overcomplicate things)